snapcraft/tests/test_sources.py

```diff
@@ -657,27 +657,51 @@ def test_init_with_source_depth_raises_exception(self):
 
 class TestLocal(tests.TestCase):
 
-    def test_pull_with_source_a_parent_of_current_dir(self):
+    def test_pull_with_source_the_parent_of_current_dir(self):
         snapcraft_files_before_pull = copy.copy(common.SNAPCRAFT_FILES)
 
         # Verify that the snapcraft root dir does not get copied into itself.
         os.makedirs('subdir')
 
-        cwd = os.getcwd()
-        os.chdir('subdir')
-        local = sources.Local('..', 'foo')
-        local.pull()
-        os.chdir(cwd)
-
-        self.assertTrue(
-            'subdir' not in os.listdir(os.path.join('subdir', 'foo')))
+        for snap_dir in [f for f in common.SNAPCRAFT_FILES if '.' not in f]:
+            cwd = os.getcwd()
+            os.chdir('subdir')
+            source_dir = os.path.join(snap_dir, 'foo')
+            local = sources.Local('..', source_dir)
+            local.pull()
+            os.chdir(cwd)
```

Comment:
this should be a `self.addCleanup(os.chdir, cwd)` and this should be right after the `os.chdir`.
I still wonder if there's a fixture we could use instead, maybe we can generalize `TempCWD` in `snapcraft.tests.fixture_setup` (if you go down this path, make it a prereq PR as it is useful in itself).